src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,411 @@
+            {
+                result += GasCostOf.TxCreate;
+            }
+
+            return result;
+        }
+        
+        [Todo(Improve.MissingFunctionality, "We need to agree on delta to state mapping details")]
+        private StateUpdate ToStateUpdate(Delta delta)
+        {
+            StateUpdate result = new StateUpdate
+            {
+                Difficulty = 1,
+                Number = 1,
+                Timestamp = (UInt256) delta.TimeStamp.Seconds,
+                GasLimit = delta.GasLimit,
+                /* here we can read coinbase entries from the delta
+                   but we need to decide how to split fees and which one to pick for the KVM */
+                GasBeneficiary = Address.Zero,
+                GasUsed = 0L
+            };
+
+            return result;
+        }
+
+        [Todo("Wider work needed to split calls and execution properly")]
+        public void CallAndRestore(Delta stateUpdate, ITxTracer txTracer, Address recipientOverride = null) { Execute(stateUpdate, txTracer, true, recipientOverride); }
+
+        [Todo("After delta is executed we should validate the state root and if it is not as expected we should revert all the changes.")]
+        public void Execute(Delta stateUpdate, ITxTracer txTracer, Address recipientOverride = null) { Execute(stateUpdate, txTracer, false, recipientOverride); }
+
+        Address GetAccountAddress(ByteString publicKeyByteString)
+        {
+            if (publicKeyByteString == null || publicKeyByteString.IsEmpty)
+            {
+                return null;
+            }
+
+            var publicKey = _cryptoContext.GetPublicKeyFromBytes(publicKeyByteString.ToByteArray());
+            return publicKey.ToKvmAddress();
+        }
+
+        private void QuickFail(ContractEntry entry, StateUpdate stateUpdate, ITxTracer txTracer, bool readOnly)
+        {
+            long gasLimit = entry.GasLimit;
+            var receiver = GetAccountAddress(entry.Base.ReceiverPublicKey);
+            var sender = GetAccountAddress(entry.Base.SenderPublicKey);
+            stateUpdate.GasUsed += gasLimit;
+            Address recipient = receiver ?? Address.OfContract(sender, _stateProvider.GetNonce(sender));
+            if (txTracer.IsTracingReceipt) txTracer.MarkAsFailed(recipient, gasLimit, Bytes.Empty, "invalid");
+        }
+        
+        private void Execute(Delta delta, ITxTracer txTracer, bool readOnly, Address recipientOverride = null)
+        {
+            // int stateSnapshot = _stateDb.TakeSnapshot();
+            // int codeSnapshot = _codeDb.TakeSnapshot();
+            // if (stateSnapshot != -1 || codeSnapshot != -1)
+            // {
+            //     if(_logger.IsError) _logger.Error($"Uncommitted state ({stateSnapshot}, {codeSnapshot}) when processing from a branch root {branchStateRoot} starting with block {suggestedBlocks[0].ToString(Block.Format.Short)}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 232 characters long line (which is greater than 200 authorized).](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)